### DISCUSSION TITLE
worldserver.exe crashing on "Initialize query data..."

### DISCUSSION THREAD

**Discussion by ayoub-khemissi** [The-Cataclysm-Preservation-Project/TrinityCore#393] [Q&A]
Hello,

At random times, when starting worldserver.exe, it crashes at "Initialize query data...". I have to retry for 10+ times to sometimes make it work and start correctly.

Here's a screenshot of the debug session with visual studio community:

![image](https://user-images.githubusercontent.com/4041328/218720342-3154355b-9f46-4fd6-81c3-2bc9c354e068.png)

We can see in the top right corner the CPU usage is increasing insanely to 100% before the crash, also something seems to be related to memory according to the autosetted breakpoint so here's my setup:

OS: Win10 professionnal x 64
Motherboard: B450 GAMING PLUS MAX (MS-7B86)
Processor: AMD Ryzen 5 3600 6-Core Processor, 3600 MHz, 6 core(s), 12 logical(s) processor(s)
RAM: 16gb 3200MHz

I've noticed this problem occured when I switched to locale frFR DBCs but I'm not really sure about this correlation.

Also, nothing appears in the Crashes folder, nore in the Server.log file.
Also I can start a 3.3.5a server without facing this problem.

Do you have any tip to fix that?

**Comment by Snify89** [comment]
Looks like your process is running out of memory? Does this occur in release mode too? I don't know much about visual studio debugging but it looks like it's allocated only 2GB
Also, is there something in your config that causes a lot of memory to load? (instances/maps/etc.?)

**Reply by ayoub-khemissi** [reply]
Hi, thank you for you answer @Snify89.

Do you have any idea of how to allocate more than 2GB? I was wondering if it was wether the cause or not. I didn't modified things verily related to processors or memory in the worldserver.conf only xp rates, server type pvp and things like that.

**Reply by Snify89** [reply]
If processes run out of memory they should raise an out of memory exception which I can not see here.
Call stack is related to the boost library. Maybe the wrong version? Not really sure tho without further information about the crash (exception)
I haven't found anything about visual studio memory limit for debugging. It is just odd, that the graph only goes up to 2GB

Maybe the community edition has (such) limits?

**Reply by ayoub-khemissi** [reply]
Starting the worldserver.exe on release mode without visual studio causes the crash too.

I'm using boost 1.73.0.

**Comment by ayoub-khemissi** [comment]
Hello,

I've found no real answer to that but the problem disappeared itself after rebooting my pc. I think it has something to do with memory or cpu usage but nothing sure.

This can be closed.
